### ISSUE TITLE
Download CPython builds with "new" CPU features (AVX)

### ISSUE THREAD

**Issue by Turakar** [astral-sh/rye#1064]
Hi, first of all thanks for the awesome project! I really enjoy the self-managed CPython versions.

I noticed that the versions downloaded do not take advantage of CPU features like AVX (SIMD), because rye always installs the `x86_64-*` variants and not the `x86_64_v{2,3,4}-*` variants, even if applicable ([doc](https://gregoryszorc.com/docs/python-build-standalone/main/running.html)). The new variants might give slight performance improvements. However, I have to note that the linked doc is not overly optimistic about the achieved performance gains, although slight gains are to be expected.

I would prefer if rye would detect the supported CPU features and download the corresponding Python version for me, as nowadays almost all common devices support at least `v3` with AVX support. I took a short look at the code, but the way rye fetches its CPython builds seems rather complex to me.

I think that this does not affect 3rd-party libraries like NumPy which ship their own AVX-supporting binaries.

**Comment by zanieb** [comment]
Hi! I believe support for this would need to be added over in https://github.com/indygreg/python-build-standalone/

**Comment by Turakar** [comment]
> Hi! I believe support for this would need to be added over in https://github.com/indygreg/python-build-standalone/

I don't think so. After all, that project already publishes corresponding builds ([e.g. on this release](https://github.com/indygreg/python-build-standalone/releases/tag/20240415)).
